src/main/kotlin/com/launchdarkly/intellij/toolwindow/FlagNode.kt

```diff
@@ -63,7 +63,7 @@ class FlagNodeVariation(private val variation: Variation) : SimpleNode() {
 
     override fun getChildren(): Array<FlagNodeBase> {
         if (variation.name != null) {
-            myChildren.add(FlagNodeBase("Value: ${variation.value}", LDIcons.DESCRIPTION))
+            myChildren.add(FlagNodeBase("Name: ${variation.name}", LDIcons.DESCRIPTION))
```

Comment:
I notice in line 76 that was deleted, there's a null check, but we aren't doing any null checks in the new code, what happens if the name is null? Does the hover documentation handle this?

Reply:
As far as I can tell, the hover documentation does not handle the null case, here is the relevant code in flagKeyHover.html, which is used by the HoverDocumentationProvider which does not null check:

`<div class="rootContainer">
    <section>
        <div>
            <span>{{ flag.name }}</span>
            <span class="flagOnOff">{{ flag.on ? "On" : "Off" }}</span>
        </div>
        {% if flag.description %}
        <div class="marginTop">
            {{ flag.description }}
        </div>
        {% endif %}
        <div class="flagDescription">
            <a href="{{ flag.url }}">Open in LaunchDarkly</a>
        </div>
    </section>
    <hr/>
    <section class="variationSection">
        <div class="heading">Variations</div>
        <ul class="variationList">
            {% for v in flag.variations %}
            <li class="variationListItem">
                <div>{{ v.value }}</div>
                {% if v.name %}
                <div class="variationName">{{ v.name }}</div>
                {% endif %}
                {% if v.description %}
                <div class="variationDescription">{{ v.description }}</div>
                {% endif %}
            </li>
            {% endfor %}
        </ul>
    </section>
</div>`

To your question, a null name here on line 66 would get skipped due to the null check in line 65, but on line 76 it references the variation value. The nullability is unknown (type Any!) so we could add a safe call (variation?.value?.toString())

Reply:
thanks for the explanation!